### PR TITLE
[telemetry] infra link information

### PR DESCRIPTION
**Pull request by superwhd**

### CODE REVIEW

**Comment by superwhd** [comment]
CI will pass when https://github.com/openthread/openthread/pull/8619 is merged.

**Review by morningboata — COMMENTED** [review]
How about add a unit test to test the DBus, maybe someplace around https://github.com/openthread/ot-br-posix/blob/main/tests/dbus/test_dbus_client.cpp

File: src/dbus/server/dbus_thread_object.cpp [hunk 1]
```diff
@@ -291,6 +293,10 @@ otbrError DBusThreadObject::Init(void)
                                std::bind(&DBusThreadObject::GetNat64ProtocolCounters, this, _1));
     RegisterGetPropertyHandler(OTBR_DBUS_THREAD_INTERFACE, OTBR_DBUS_PROPERTY_NAT64_ERROR_COUNTERS,
                                std::bind(&DBusThreadObject::GetNat64ErrorCounters, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_THREAD_INTERFACE, OTBR_DBUS_PROPERTY_NAT64_ERROR_COUNTERS,
+                               std::bind(&DBusThreadObject::GetNat64ErrorCounters, this, _1));
```
Comment (morningboata) [inline src/dbus/server/dbus_thread_object.cpp:297]: Remove this line?

**Comment by superwhd** [comment]
> How about add a unit test to test the DBus, maybe someplace around https://github.com/openthread/ot-br-posix/blob/main/tests/dbus/test_dbus_client.cpp

I'll verify this feature in a border router test case in openthread repo.

**Review by morningboata — APPROVED** [review]

**Review by librasungirl — APPROVED** [review]

File: src/dbus/common/types.hpp [hunk 2]
```diff
@@ -690,6 +690,17 @@ struct Nat64ErrorCounters
     Nat64PacketCounters mNoMapping;        ///< Packet drop due to no mappings found or mapping pool exhausted.
 };
 
+struct InfraLinkInfo
+{
+    std::string mName;                   ///< The name of the infrastructure network interface
```
Comment (jwhui) [inline src/dbus/common/types.hpp:695]:
```suggestion
    std::string mName;                   ///< The name of the infrastructure network interface.
```

File: src/dbus/server/introspect.xml [hunk 3]
```diff
@@ -893,6 +893,23 @@
       <annotation name="org.freedesktop.DBus.Property.EmitsChangedSignal" value="false"/>
     </property>
 
+    <!-- InfraLinkInfo: Information of infrastructure network interface.
+    <literallayout>
+        struct {
+          string name;                     // The name of the infrastructure network interface
```
Comment (jwhui) [inline src/dbus/server/introspect.xml:899]:
```suggestion
          string name;                     // The name of the infrastructure network interface.
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**